crates/protocol/src/lib.rs

```diff
@@ -1,2 +1,6 @@
 pub mod message;
 pub mod resource;
+pub mod resources;
```

Comment:
So help me understand the way namespacing works in Rust. I see we have both a `resource` (singular) and a `resources` (plural), and I can see the reasoning, but how would this work in practice for the app-devs? Like if they wanted to use `ResourceMetadata` how is that imported vs if they wanted to use `Offering`?

I'm cool with it, but worth exploring the alternative of having only a `resources` namespace, and moving the contents within `resource.rs` inside of it, albeit maybe in a differently named file? Idk I'm a little out of my league here in Rust land.

Reply:
I was hoping someone might ask about this! I've been torn about how to best setup our Rust repos, and just chose something to get over my blank-page syndrome. I am fully open to changes here.

In practice, an app developer would currently consume resource/resources with the following:
* `use protocol::resource::{Resource, ResourceMetadata, ResourceError}`
   * This is how a user would pull in a base generic Resource, Metadata, and Errors into their own Rust code
* `use protocol::resources::offering::{Offering, OfferingData}`
   * This is how a user would pull in an Offering (which constructs a concrete `Resource<OfferingData>`) and the OfferingData itself
   
---

We could combine these into the same "use path" by structuring it like either of the following folder strctures:
```
Option 1                          Option 2
src                               src
|_ resource.rs                    |_ resource
|_ resource                           |_ mod.rs
    |_ offering.rs                    |_ offering.rs
```
Both of these would produce the following use paths:
```
use protocol::resource::{Resource, ResourceMetadata, ResourceError}
use protocol::resource::offering::{Offering, OfferingData}
```
The only difference would be that in Option 2, the current contents of `resource.rs` would be moved into the special `mod.rs` file.

Reply:
Okay cool cool

So, I vote we do standardize the namespace to be consistent, be that either `resource` (singular) or `resources` (plural). I can't explain why, but I tend to prefer plural, but it's bikeshedding so idc just pick one. But this...

```rust
use protocol::resource::{Resource, ResourceMetadata, ResourceError}
use protocol::resource::offering::{Offering, OfferingData}
```

...or this...

```rust
use protocol::resources::{Resource, ResourceMetadata, ResourceError}
use protocol::resources::offering::{Offering, OfferingData}
```

...is a DX much preferred to this...

```rust
use protocol::resource::{Resource, ResourceMetadata, ResourceError}
use protocol::resources::offering::{Offering, OfferingData}
```

---

As for which file/folder structure to use, Option 1 or 2, just pick whichever feels more intuitive, I don't have any preference on that matter.

Reply:
For reference [KT implementation](https://github.com/TBD54566975/tbdex-kt/tree/main/protocol/src/main/kotlin/tbdex/sdk/protocol/models) is all under the `models` namespace, but I much much much prefer your namespacing here, so IMO we should do the same over in `tbdex-kt`

[In TS/JS](https://github.com/TBD54566975/tbdex-js/tree/main/packages/protocol/src), namespacing isn't really much of a thing, so it's not as applicable.

Reply:
Perfect, and I agree that it's a preferred DX experience! As I was typing out the words explaining it, I was like "oh damn, this _is_ better."

I'll go with Option 2 for now, as it brings all the "Resource" code within the same folder!